buildSrc/build.gradle

```diff
@@ -199,24 +199,21 @@ if (project != rootProject) {
     into localDownloads
   }
 
-  unitTest {
```

Comment:
I'm not sure all projects having a `test` task that does nothing, in addition to a `unitTest` task is particularly clear. I'm not sure we should continue with this funny convention just for historical purposes. I don't think educating folks that `:test` == unit tests is asking a lot.

Reply:
The change wasn't done only as a technical limitations. 
I do and always have hated the Gradle `test` task for the ambiguity it brings, 
and for running `IT` by default, it makes it too easy to mangle tests of different kinds, and it really probably only exists for historical reasons because other build tools had it.
Also, the fact that it's being changed in this PR has nothing to do with the goal of the PR and just makes it much harder to read.   

My goal is to have specific `unitTest`, `integTest`, and `internalClusterTest` tasks with well defined meanings eventually. `test` doesn't contribute to this clarity,
it's just a catch all which makes it seem ok to add stuff there and move on, whereas `unitTest` makes one think about  what tests are being added and what's the best place for them. 

Gradle is broken in that it enforces a testing conventions that does not scale for anything but projects of trivial size with no convenient way to opt-out.

Reply:
> I do and always have hated the Gradle test task for the ambiguity it brings,
and for running IT by default, it makes it too easy to mangle tests of different kinds,

To be fair, I think you are just describing Maven surefire and failsafe plugin behaviors. Which themselves are a workaround to a technical limitation which is that Maven doesn't support more than two source sets per module. The _proper_ way to separate tests like this is to model them as separate source sets, not just different `Test` tasks with includes/excludes based on naming conventions. This has lots of other benefits as well.

> My goal is to have specific unitTest, integTest, and internalClusterTest tasks with well defined meanings eventually.

I think we are bikeshedding to some degree. Again, I don't think it's a big deal to convey that `test` runs unit tests, and other tests run the tests as the name describes.

> Gradle is broken in that it enforces a testing conventions that does not scale for anything but projects of trivial size with no convenient way to opt-out.

I think it's the opposite actually. It _doesn't_ enforce any conventions, which I think is your complaint. I think the "convention" you have issue with is the name of the task it creates. This is really the only thing Gradle is "forcing" on us.